kyo-tag/shared/src/main/scala/kyo/tags.scala

```diff
@@ -256,29 +266,37 @@ object Tag:
                     case tpe: OrType   => report.errorAndAbort(s"Intersection tags don't support type unions. Found: ${tpe.show}")
                     case tpe           => Seq(tpe)
 
-            '{ Intersection(${ tags(using q)(flatten) }) }
+            '{ Intersection.raw[T](${ tags(using q)(flatten) }) }
         end intersectionImpl
 
-        // printable ASCII chars
-        val maxEncodableInt = ('~' - ' ').toInt
+        // Encodes ints using printable ASCII chars
+        val charsReserved  = immutable.Set('[', ']', '(', ')', ';', ',')
```

Comment:
I found a bug while working on this change: characters that are used to control the subtype check can end up encoded as the size or hash of segment entries, breaking the logic. This new mechanism ensures the reserved chars aren't used to represent `Int`s.